### PR TITLE
Added the ability to exportGame via api

### PR DESCRIPTION
**Pull request by phyzical**
* added the ability to check if there is a bonus avaible via the api
* Added the ability to exportGame via api

# Feature
tested exportBonus before exporting returns true
tested exportBonus after triggering an export returns false
tested exportGame triggers the same functionality as the ui

### CODE REVIEW

File: src/NetscriptFunctions/Singularity.ts [hunk 1]
```diff
@@ -250,7 +252,12 @@ export function NetscriptSingularity(
       workerScript.running = false; // Prevent workerScript from "finishing execution naturally"
       killWorkerScript(workerScript);
     },
-
+    backupSave: function (): void {
+      return exportGame(player)
+    },
+    backupSaveBonus: function (): boolean {
+      return canGetBonus();
+    },
```
Comment (nickofolas) [inline src/NetscriptFunctions/Singularity.ts:260]: These should probably call
```ts
helper.updateDynamicRam("backupSave", getRamCost(player, "backupSave"));
helper.checkSingularityAccess("backupSave")
```
and same for `backupSaveBonus`, to keep in line with the other Singularity functions.

**Review by nickofolas — CHANGES_REQUESTED** [review]
Looks good, just one small suggestions for the Singularity access.

File: src/NetscriptFunctions/Singularity.ts (hunk 1 above)
Comment (phyzical) [reply src/NetscriptFunctions/Singularity.ts:260]: ah yep! oops good spot 👍
i've never added a new function that needed ram

**Comment by phyzical** [comment]
not sure why it fails, locally it does not fail to build...

**Pull request merged**